### ISSUE TITLE
Native crash "Error, cannot access an invalid/free'd bitmap here!"

### ISSUE THREAD

**Issue by jeffbordon** [mapsforge/mapsforge#957]
After a few fast rotations of the map I encounter a native crash that is hard to trace back.
I found out that this issue occurs since I have updated the `MapViewerTemplate` to the latest version which calls `AndroidGraphicFactory.clearResourceMemoryCache()` in `onDestroy()`.

Any ideas how to fix this?

> W/Bitmap: Called getWidth() on a recycle()'d bitmap! This is undefined behavior!
> A/Bitmap: Error, cannot access an invalid/free'd bitmap here!
> A/libc: Fatal signal 6 (SIGABRT), code -6 in tid 21149 (pool-22-thread-)
> 
> 05-13 13:47:26.804 1504-1504/? A/DEBUG: Abort message: 'Error, cannot access an invalid/free'd bitmap here!'
> 05-13 13:47:26.804 1504-1504/? A/DEBUG:     r0 00000000  r1 0000529d  r2 00000006  r3 9a73f978
> 05-13 13:47:26.804 1504-1504/? A/DEBUG:     r4 9a73f980  r5 9a73f930  r6 00000058  r7 0000010c
> 05-13 13:47:26.804 1504-1504/? A/DEBUG:     r8 aeddedb8  r9 9ef1dd00  sl ffffffff  fp 1340d640
> 05-13 13:47:26.804 1504-1504/? A/DEBUG:     ip 00000006  sp 9a73d718  lr b6cc1b61  pc b6cc3f50  cpsr 400f0010
> 05-13 13:47:26.817 1504-1504/? A/DEBUG: backtrace:
> 05-13 13:47:26.818 1504-1504/? A/DEBUG:     #00 pc 00041f50  /system/lib/libc.so (tgkill+12)
> 05-13 13:47:26.818 1504-1504/? A/DEBUG:     #01 pc 0003fb5d  /system/lib/libc.so (pthread_kill+32)
> 05-13 13:47:26.818 1504-1504/? A/DEBUG:     #02 pc 0001c30f  /system/lib/libc.so (raise+10)
> 05-13 13:47:26.818 1504-1504/? A/DEBUG:     #03 pc 000194c1  /system/lib/libc.so (__libc_android_abort+34)
> 05-13 13:47:26.818 1504-1504/? A/DEBUG:     #04 pc 000174ac  /system/lib/libc.so (abort+4)
> 05-13 13:47:26.819 1504-1504/? A/DEBUG:     #05 pc 0000864b  /system/lib/libcutils.so (__android_log_assert+86)
> 05-13 13:47:26.819 1504-1504/? A/DEBUG:     #06 pc 0008d5b7  /system/lib/libandroid_runtime.so (_ZNK7android6Bitmap11assertValidEv+18)
> 05-13 13:47:26.819 1504-1504/? A/DEBUG:     #07 pc 0008d633  /system/lib/libandroid_runtime.so (_ZN7android6Bitmap11getSkBitmapEP8SkBitmap+6)
> 05-13 13:47:26.819 1504-1504/? A/DEBUG:     #08 pc 0008b285  /system/lib/libandroid_runtime.so
> 05-13 13:47:26.819 1504-1504/? A/DEBUG:     #09 pc 73436db9  /data/dalvik-cache/arm/system@framework@boot.oat (offset 0x1ed6000)

**Comment by devemux86** [comment]
Could be a race condition due to fast exit / restart.

Calling `onDestroy` when rotating the device means that the activity (i.e the app) is finished and recreated with each orientation change.
Probably not very nice for performance or user experience, why not avoiding it?
e.g. we have disabled it in mapsforge-samples-android [manifest](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-samples-android/AndroidManifest.xml):
```xml
android:configChanges="keyboardHidden|orientation|screenSize"
```

**Comment by jeffbordon** [comment]
Thanks! I already thought about this option since it seems the easiest one. However, it feels like misusing the Android framework to break the activity lifecycle. I will try to narrow the issue down further and update you when I identify the issue.

**Comment by jeffbordon** [comment]
I was able to reproduce the issue in the simplest `DefaultTheme` example by just removing `android:configChanges="keyboardHidden|orientation|screenSize"` from the manifest.

As a temporary workaround I removed the call to `AndroidGraphicFactory.clearResourceMemoryCache()` from `onDestroy()` so the issue does not appear (however, this might not be very memory efficient).

From my basic understanding the Bitmaps are recycled in `AndroidGraphicFactory.clearResourceMemoryCache()` while they are still accessed from somewhere else (e.g. the tile renderer) which leads to the native crash.
